pvlib/irradiance.py

```diff
@@ -3117,3 +3117,72 @@ def complete_irradiance(solar_zenith,
                                      'dhi': dhi,
                                      'dni': dni})
     return component_sum_df
+
+
+def pvl_louche(ghi, solar_zenith, datetime_or_doy):
+    """
+    Determine DNI and GHI from GHI using Louche model.
+
+    Parameters
+    ----------
+    ghi : Series
+        Global horizontal irradiance. [W/m^2]
+
+    solar_zenith : Series
+        True (not refraction-corrected) zenith angles in decimal
+        degrees. Angles must be >=0 and <=90.
+
+    datetime_or_doy : numeric, pandas.DatetimeIndex
+        Day of year or array of days of year e.g.
+        pd.DatetimeIndex.dayofyear, or pd.DatetimeIndex.
+
+    Returns
+    -------
+    data: OrderedDict or DataFrame
+        Contains the following keys/columns:
+
+            * ``dni``: the modeled direct normal irradiance in W/m^2.
+            * ``dhi``: the modeled diffuse horizontal irradiance in
+              W/m^2.
+            * ``kt``: Ratio of global to extraterrestrial irradiance
+              on a horizontal plane.
+    Notes
+    -----
+    `solar_zenith` angles should be between 0 and 90 degrees else NaN will be
+    returned. Day of year (doy) should be between 1 and 366 else NaN will be
+    returned.
+
+    References
+    -------
+    .. [1] Louche A, Notton G, Poggi P, Simmonnot G. Correlations for direct
+       normal and global horizontal irradiation on French Mediterranean site.
+       Solar Energy 1991;46:261-6
+
+    """
+    bool = np.logical_or(solar_zenith > 90, solar_zenith < 0)
+    solar_zenith = np.where(bool, np.NaN, solar_zenith)
```

Comment:
For consistency with other decomposition functions, I think we should return zero (not `nan`) when `solar_zenith > 90`.  I also don't think checking `solar_zenith < 0` is necessary.  I suggest replacing this code with the same `bad_values` check used in `boland`, `erbs`, `disc` etc.

Reply:
I have removed this check and added the bad_values check of ```boland```,```erbs```,```disc``` . In test cases I had added one test case for negative zenith, I removed it but I think the default checks function should be updated to accommodate this possibility.